analytics/src/analytics/integrations/github/validation.py

```diff
@@ -0,0 +1,119 @@
+"""Pydantic schemas for validating GitHub API responses."""
+
+# pylint: disable=no-self-argument
+from datetime import datetime, timedelta
+
+from pydantic import BaseModel, Field, computed_field, model_validator
+
+
+def safe_default_factory(data: dict, keys_to_replace: list[str]) -> dict:
+    """
+    Replace keys that are explicitly set to None with an empty dict for default_factory.
+
+    We need to do this because if a key is present, but its value is None or null,
+    it will raise a Pydantic ValidationError rather than using the default_factory.
+    """
+    for key in keys_to_replace:
+        if data.get(key) is None:
+            data[key] = {}
+    return data
+
+
+# #############################################
+# Issue content sub-schemas
+# #############################################
+
+
+class IssueParent(BaseModel):
+    """Schema for the parent issue of a sub-issue."""
+
+    title: str | None = None
+    url: str | None = None
+
+
+class IssueType(BaseModel):
+    """Schema for the type of an issue."""
+
+    name: str | None = None
+
+
+class IssueContent(BaseModel):
+    """Schema for core issue metadata."""
+
+    title: str
+    url: str
+    closed: bool
+    created_at: str = Field(alias="createdAt")
+    closed_at: str | None = Field(alias="closedAt", default=None)
+    issue_type: IssueType = Field(alias="type", default_factory=IssueType)
+    parent: IssueParent = Field(default_factory=IssueParent)
+
+    @model_validator(mode="before")
+    def replace_none_with_defaults(cls, values) -> dict:  # noqa: ANN001, N805
+        """Replace None with default_factory instances."""
+        # Replace None with default_factory instances
+        return safe_default_factory(values, ["type", "parent"])
```

Comment:
Here's where we apply that `safe_default_factory()` to catch and replace the keys for which we want to use the default factory if they're explicitly set to `None`